lib/plugins.ts

```diff
@@ -5,6 +5,21 @@ export interface ILocalStoragePluginState<S extends IState> {
   __version: number
 }
 
+const migrateState = <S extends IState>(oldState: IState, newState: S): S => {
```

Comment:
This is the function is used to migrate old state to new state.
It is expected to compare the previous fields to the new ones, removes deleted fields, and sets defaults for added fields.